compiler/rustc_infer/src/infer/error_reporting/nice_region_error/find_anon_type.rs

```diff
@@ -145,23 +137,30 @@ impl<'tcx> Visitor<'tcx> for FindNestedTypeVisitor<'tcx> {
             }
             // Checks if it is of type `hir::TyKind::Path` which corresponds to a struct.
             hir::TyKind::Path(_) => {
-                let subvisitor = &mut TyPathVisitor {
-                    tcx: self.tcx,
-                    found_it: false,
-                    bound_region: self.bound_region,
-                    current_index: self.current_index,
+                // Prefer using the lifetime in type arguments rather than lifetime arguments.
+                intravisit::walk_ty(self, arg)?;
```

Comment:
It looks like we have an extra call to `walk_ty` here? I don't see where this was happening in the old version of the code.

Reply:
That match arm used to continue to the `walk_ty` call at the end of the function.